backend/bitsurance/bitsurance.go

```diff
@@ -35,70 +33,38 @@ const (
-// For details on the hashing please refer to `BitsuranceGetId`.
-func fetchBitsuranceIds(httpClient *http.Client) ([]string, error) {
-	endpoint := apiURL + "allAccountHashes"
-	req, err := http.NewRequest(http.MethodGet, endpoint, nil)
-	if err != nil {
-		return nil, errp.WithStack(err)
-	}
-	req.Header.Add("X-API-KEY", apiKey)
+const statusActive = "active"
 
-	res, err := httpClient.Do(req)
-	if err != nil {
-		return nil, errp.WithStack(err)
-	}
-	defer res.Body.Close() //nolint:errcheck
-	if res.StatusCode != http.StatusOK {
-		return nil, errp.Newf("%s - bad response code %d", endpoint, res.StatusCode)
-	}
-	const max = 81920
-	responseBody, err := io.ReadAll(io.LimitReader(res.Body, max+1))
-	if err != nil {
-		return nil, errp.WithStack(err)
-	}
-	return strings.Split(string(responseBody), "\n"), nil
+type accountDetails struct {
+	Status string `json:"status"`
 }
 
-// findInsuredAccounts compares `accountIds` and `insuredIds` and returns a slice
-// with the matching elements.
-//
-// `accountIds` is an ordered slice of strings, containing the hashes of the
-// requested wallet accounts xpubs
-// `insuredIds` is another ordered slice of strings, containing the hashes of
-// all the insured xpubs retrieved from the Bitsurance server.
+// bitsuranceCheckId fetches the account details of the passed accountId from the
+// Bitsurance server. It returns true if the account status is "active". Possible
+// status are (from Bitsurance Api docs):
+// - "processing"	when still in creation/checking phase (short time)
+// - "refused" application got refused
+// - "waitpayment" accepted, but waiting on first payment
+// - "active" insurance coverage activated
+// - "inactive" offer support link for more info (maybe missed payment)
+// - "canceled" contract got canceled
 //
-// The complexity of the scanning is linear.
-// For details on the hashing please refer to `BitsuranceGetId`.
-// TODO add tests.
-func findInsuredAccounts(accountIds, insuredIds []string) []string {
-	var matching []string
-
-	// Initialize indices for both lists
-	i, j := 0, 0
-
-	// Traverse both lists while there are elements in both
-	for i < len(accountIds) && j < len(insuredIds) {
-		// Compare the current elements in both lists
-		switch {
-		case accountIds[i] == insuredIds[j]:
-			// If the elements match, add it to the matching slice
-			matching = append(matching, accountIds[i])
-			i++
-			j++
-		case accountIds[i] < insuredIds[j]:
-			// If the element in accountIds is smaller, move to the next element
-			i++
-		default:
-			// If the element in insuredIds is smaller, move to the next element
-			j++
-		}
+// If an accountId is not retrieved at all, the endpoint return a 404 http code.
+func bitsuranceCheckId(httpClient *http.Client, accountId string) (bool, error) {
+	endpoint := apiURL + "accountDetails/" + accountId
+	details := accountDetails{}
+	code, err := util.APIGet(httpClient, endpoint, apiKey, 1024, &details)
+	if *code == http.StatusNotFound {
```

Comment:
do this check after `err != nil` check or this can panic

Reply:
nice catch! Thanks